ksql-serde/src/test/java/io/confluent/ksql/serde/avro/KsqlGenericRowAvroDeserializerTest.java

```diff
@@ -105,4 +118,121 @@ public void shouldCreateCorrectRow() {
         (100.0));
   }
 
+  @Test
+  public void shouldDeserializeCorrectly() {
+    SchemaRegistryClient schemaRegistryClient = new MockSchemaRegistryClient();
+    List columns = Arrays.asList(1511897796092l, 1l, "item_1", 10.0, new Double[]{100.0},
+                                 Collections.singletonMap("key1", 100.0));
+
+    GenericRow genericRow = new GenericRow(columns);
+
+    KsqlGenericRowAvroDeserializer ksqlGenericRowAvroDeserializer = new
+        KsqlGenericRowAvroDeserializer(schema, schemaRegistryClient, false);
+
+    byte[] serializedRow = getSerializedRow("t1", schemaRegistryClient, avroSchema, genericRow);
+
+    GenericRow row = ksqlGenericRowAvroDeserializer.deserialize("t1", serializedRow);
+    Assert.assertNotNull(row);
+    assertThat("Incorrect deserializarion", row.getColumns().size(), equalTo(6));
+    assertThat("Incorrect deserializarion", (Long)row.getColumns().get(0), equalTo(1511897796092l));
```

Comment:
you don't need the casts to Long.

also, use `L` instead of `l` for longs as the lower case version is too easy to mistake as a 1

Reply:
Done